op-e2e/actions/garbage_channel_out.go

```diff
@@ -0,0 +1,246 @@
+package actions
+
+import (
+	"bytes"
+	"compress/gzip"
+	"compress/zlib"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type GarbageKind int64
+
+const (
+	STRIP_VERSION GarbageKind = iota
+	RANDOM
+	TRUNCATE_END
+	DIRTY_APPEND
+	INVALID_COMPRESSION
+	MALFORM_RLP
+)
+
+var GarbageKinds = []GarbageKind{
+	STRIP_VERSION,
+	RANDOM,
+	TRUNCATE_END,
+	DIRTY_APPEND,
+	INVALID_COMPRESSION,
+	MALFORM_RLP,
+}
+
+// GarbageChannelCfg is the configuration for a `GarbageChannelOut`
+type GarbageChannelCfg struct {
+	useInvalidCompression bool
+	malformRLP            bool
+}
+
+// WriterApi is the interface shared between `zlib.Writer` and `gzip.Writer`
+type WriterApi interface {
+	Close() error
+	Flush() error
+	Reset(io.Writer)
+	Write([]byte) (int, error)
+}
+
```

Comment:
Prob want to add a compile time check that it satisfies the interface, like so: https://medium.com/@matryer/golang-tip-compile-time-checks-to-ensure-your-type-satisfies-an-interface-c167afed3aae

Could also do the same thing for the actual impl in `op-node`